clang/test/SemaSYCL/sycl-device-num_simd_work_items-template.cpp

```diff
@@ -93,3 +93,72 @@ int check() {
 // CHECK-NEXT: ConstantExpr {{.*}} 'int'
 // CHECK-NEXT: value: Int 2
 // CHECK-NEXT: IntegerLiteral{{.*}}2{{$}}
+
+// Tests for num_simd_work_items and reqd_work_group_size arguments check.
+template <int N>
+__attribute__((reqd_work_group_size(3, 6, 4))) void func6(); // expected-note{{conflicting attribute is here}}
+template <int N>
+[[intel::num_simd_work_items(N)]] void func6(); // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+
+template <int N>
+[[cl::reqd_work_group_size(8, 4, 5)]] void func7(); // expected-note{{conflicting attribute is here}}
+template <int N>
+[[intel::num_simd_work_items(N)]] void func7(); // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+
+template <int N>
+[[intel::reqd_work_group_size(N, N, N)]] void func8(); // expected-note{{conflicting attribute is here}}
+template <int N>
+[[intel::num_simd_work_items(3)]] void func8(); // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+
+template <int X, int Y, int Z, int N>
+[[intel::reqd_work_group_size(X, Y, Z)]] void func9(); // expected-note{{conflicting attribute is here}}
+template <int X, int Y, int Z, int N>
+[[intel::num_simd_work_items(N)]] void func9(); // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+
+template <int X, int Y, int Z>
+[[intel::reqd_work_group_size(X, Y, Z)]] void func10(); // expected-note{{conflicting attribute is here}}
+template <int X, int Y, int Z>
+[[intel::num_simd_work_items(3)]] void func10(); // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+
+template <int X, int Y, int Z>
+[[cl::reqd_work_group_size(X, Y, Z)]] void func11(); // expected-note{{conflicting attribute is here}}
+template <int X, int Y, int Z>
+[[intel::num_simd_work_items(2)]] void func11(); // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+
+template <int N>
+[[cl::reqd_work_group_size(N, N, N)]] void func12(); // expected-note{{conflicting attribute is here}}
+template <int N>
+[[intel::num_simd_work_items(2)]] void func12(); // expected-error{{'num_simd_work_items' attribute must evenly divide the work-group size for the 'reqd_work_group_size' attribute}}
+
+int check1() {
+  // no error expected
+  func6<3>();
+  //expected-note@+1{{in instantiation of function template specialization 'func6<2>' requested here}}
```

Comment:
FWIW, I usually prefer putting the `expected-whatever` line on the line that's impacted rather than using the `@+1` syntax. It tends to make it more immediately obvious what's being diagnosed.

Reply:
Done